packages/headless/src/composables/listbox/createListbox.ts

```diff
@@ -1,34 +1,146 @@
-import { computed, unref, type MaybeRef } from "vue";
+import { computed, ref, unref, watchEffect, type MaybeRef, type Ref } from "vue";
+import { createId } from "../..";
 import { createBuilder } from "../../utils/builder";
 
 export type CreateListboxOptions = {
   /**
    * Aria label for the listbox.
    */
   label: MaybeRef<string>;
+  /**
+   * Value of currently selected option.
+   */
+  selectedOption: Ref<ListboxValue | undefined>;
+  /**
+   * Value of currently (visually) focused option.
+   */
+  focusedOption: Ref<ListboxValue | undefined>;
   /**
    * Whether the listbox is multiselect.
    */
   multiselect?: MaybeRef<boolean | undefined>;
   /**
    * Hook when an option is selected.
    */
-  onSelect?: (id: ListboxValue) => void;
+  onSelect?: (value: ListboxValue) => void;
+  /**
+   * Hook when an option should be scrolled into view.
+   */
+  onScrollIntoView?: (id: string, value: ListboxValue) => void;
+  /**
+   * Hook when the first option should be focused.
+   */
+  onFocusFirst?: () => void;
+  /**
+   * Hook when the last option should be focused.
+   */
+  onFocusLast?: () => void;
+  /**
+   * Hook when the next option should be focused.
+   */
+  onFocusNext?: (currentValue: ListboxValue) => void;
+  /**
+   * Hook when the previous option should be focused.
+   */
+  onFocusPrevious?: (currentValue: ListboxValue) => void;
+  /**
+   * Hook when the first option starting with the given label should be focused.
+   */
+  onFocusByLabel?: (label: string) => void;
```

Comment:
“Focus” is an imprecise term here. The real focus always stays on the listbox itself. So IMHO would be preferable to either drop the term “focus” or use the term “active”, as that is what is used in the spec: https://w3c.github.io/aria/#managingfocus

Reply:
Done